### ISSUE TITLE
expectedFailure() decorator

### ISSUE THREAD

**Issue by brianjmurrell** [avocado-framework/avocado#2933]
As [suggested on the ML](https://www.redhat.com/archives/avocado-devel/2018-November/msg00050.html), I'm opening an issue here to support `unittest`'s `expectedFailure()` decorator.

**Comment by clebergnu** [comment]
Hi @brianjmurrell,

I was thinking about this feature and come up with a few things.  Python's own unittest implements a different status, that is:

```
x
----------------------------------------------------------------------
Ran 1 test in 0.000s

OK (expected failures=1)
```

It clearly marks the test as an expected failure.  At the same time that I understand the motivation, I get back into thinking that this means that we got what we wanted out of the test (and some would say that this means the test PASSed).

How do you see it?  Do you have real world use cases to contribute?

**Comment by brianjmurrell** [comment]
Yes, as I understand it, you use `expectedFailure()` to make your test be considered a pass if a test _fails_.  Negative testing I guess.

I must have had a real world use case when I filed this, but it escapes me now.

**Comment by willianrampazzo** [comment]
Hi @brianjmurrell, this issue was opened more than a year ago. Is this feature still needed?
